File: app/src/main/java/com/sunzk/colortest/Constant.java
package com.sunzk.colortest;

import android.content.Intent;

import com.sunzk.colortest.activity.FindDiffColorActivity;
import com.sunzk.colortest.activity.GuessColorActivity;
import com.sunzk.colortest.activity.MockColorActivity;
import com.sunzk.colortest.activity.SelectPicActivity;
import com.sunzk.colortest.entity.ModeEntity;

import static com.sunzk.colortest.activity.GuessColorActivity.INTENT_KEY_DIFFICULTY;

public class Constant {
	
	public static final ModeEntity[] MODE_ENTITY_LIST;
	public static final String MODE_SELECT_DATA_NAME = "settings";
	public static final String MODE_SELECT_DATA_KEY = "modeList";
	
	static {
		MyApplication context = MyApplication.getInstance();
		
		Intent hellIntent1 = new Intent(context, GuessColorActivity.class);
		hellIntent1.putExtra(INTENT_KEY_DIFFICULTY, GuessColorActivity.Difficulty.EASY);
		Intent hellIntent2 = new Intent(context, GuessColorActivity.class);
		hellIntent1.putExtra(INTENT_KEY_DIFFICULTY, GuessColorActivity.Difficulty.MEDIUM);
		Intent hellIntent3 = new Intent(context, GuessColorActivity.class);
		hellIntent1.putExtra(INTENT_KEY_DIFFICULTY, GuessColorActivity.Difficulty.DIFFICULT);

		MODE_ENTITY_LIST = new ModeEntity[]{
			new ModeEntity("标准模式", new Intent(context, MockColorActivity.class)),
			new ModeEntity("地狱模式(体验版)", hellIntent1),
			new ModeEntity("地狱模式(中等)", hellIntent2), 
			new ModeEntity("地狱模式(困难)", hellIntent3),
			new ModeEntity("找不同", new Intent(context, FindDiffColorActivity.class)),
			new ModeEntity("选图片", new Intent(context, SelectPicActivity.class))
		};

	}
	
}

File: app/src/main/java/com/sunzk/colortest/Constant.kt
package com.sunzk.colortest

import com.sunzk.colortest.findDiff.FindDiffColorActivity
import com.sunzk.colortest.intermediateColor.IntermediateColorActivity
import com.sunzk.colortest.mockColor.MockColorActivity
import com.sunzk.colortest.entity.GameMode
import com.sunzk.colortest.entity.ModeEntity

object Constant {

	val BASE_URL_GITHUB = "https://api.codetabs.com/v1/proxy/?quest=https://raw.githubusercontent.com/sunzhk/ColorTest/data/"
	val BASE_URL_MOCK = "http://mock-api.com/rnNrALnl.mock/"

	const val SP_NAME_APP = "AppSetting"
	const val SP_KEY_NIGHT_MODE = "Key_NightMode"
	const val SP_KEY_BGM = "Key_BGM"
	
	val MODE_ENTITY_LIST: Array<ModeEntity>
	const val MODE_SELECT_DATA_NAME = "settings"
	const val MODE_SELECT_DATA_KEY = "modeList"

	init {
		MODE_ENTITY_LIST = arrayOf(
			ModeEntity("标准模式", GameMode.MockColor, MockColorActivity::class.java.name),
			ModeEntity("找中间色", GameMode.IntermediateColor, IntermediateColorActivity::class.java.name),
			ModeEntity("找不同", GameMode.FindDiffColor, FindDiffColorActivity::class.java.name),
//			ModeEntity("选图片", SelectPicActivity::class.java.name)
		)
	}
}